File: python/allergies/allergies.py
class Allergies:

    def __init__(self, score):
        self.score = score % 256
        self.allergens = {
            'eggs': 1,
            'peanuts': 2,
            'shellfish': 4,
            'strawberries': 8,
            'tomatoes': 16,
            'chocolate': 32,
            'pollen': 64,
            'cats': 128
        }

    def allergic_to(self, item):
        return self.score & self.allergens[item] != 0

    @property
    def lst(self):
        return [allergen for allergen in self.allergens.keys() if self.allergic_to(allergen)]

File: ruby/allergies/allergies.rb
class Allergies
  ALLERGENS = {
    'eggs' => 1,
    'peanuts' => 2,
    'shellfish' => 4,
    'strawberries' => 8,
    'tomatoes' => 16,
    'chocolate' => 32,
    'pollen' => 64,
    'cats' => 128
  }

  def initialize(score)
    @score = score % 256
  end

  def allergic_to?(allergen)
    @score & ALLERGENS[allergen] != 0
  end

  def list
    allergies = []
    ALLERGENS.each { |k, _|
      allergic_to?(k) and allergies.push k
    }
    allergies
  end
end
